File: Exercises/Week02-SOLID-Java/src/week2sensors/ControlUnit.java
package week2sensors;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by nathanhanak on 1/19/17.
 */

public class ControlUnit {

    public void pollSensors() {
        List<Sensor> sensors = new ArrayList<Sensor>();
        sensors.add(new FireSensor("Basement"));
        sensors.add(new SmokeSensor("Auditorium"));

        for (Sensor sensor : sensors) {
            if (sensor.isTriggered()) {
                System.out.println("A " + sensor.getSensorType() + " sensor was triggered at " + sensor.getLocation());
            } else {
                System.out.println("Polled " + sensor.getSensorType() + " at " + sensor.getLocation() + " successfully");
            }
        }
    }
}

File: Exercises/Week02-SOLID-Scala/src/week2sensors/ControlUnit.scala
package week2sensors

import scala.collection.mutable.ArrayBuffer

/**
  * Created by nathanhanak on 1/19/17.
  */
class ControlUnit {

  def pollSensors(): Unit = {
    val fs1 : Sensor = new FireSensor("Basement")
    val ss1 : Sensor = new SmokeSensor("Auditorium")
    val sensors = ArrayBuffer(fs1, ss1)

    for ( s <- sensors ) {
      if (s.isTriggered() ) println(s"A ${s.getSensorType()} sensor was triggered at ${s.getLocation()}")
      else println("Polled ${s.getSensorType} at ${s.getLocation} successfully")
    }
  }

}


/** #3. Currently, pollSensors responsibilities are to create a list of sensors, fill it, and then check
       if they've been triggered

  */